### PR TITLE
improve authoring interactive sizing

### PR DESCRIPTION
**Pull request by scytacki**
it now handles aspect ratio better
Note this is still not the same handling as at runtime,
it doesn't handle the 'MAX' setting, limit the height, or change depending on the context

### CODE REVIEW

**Comment by scytacki** [comment]
Sorry I didn't realize this was half baked. I've got a to update the large FIXME comment.

**Comment by scytacki** [comment]
@dougmartin ok this is ready to look at now

File: lara-typescript/src/page-item-authoring/common/components/interactive-iframe.tsx [hunk 1]
```diff
@@ -25,13 +21,38 @@ interface IFramePhoneParentEndpoint {
...
+  // So probably the best option is to abstract that sizing code so it can be
+  // used by both jQuery and React
+  const [height, setHeight] = useState<number|string|null>(null);
+  const [aspectRatio, setAspectRatio] = useState<number>(authoredAspectRatio);
+
+  const handleHeightChange = (newHeight: number | string) => {
+    setHeight(newHeight);
+  };
+
+  const handleSupportedFeatures = (info: any) => {
```
Comment (dougmartin) [inline lara-typescript/src/page-item-authoring/common/components/interactive-iframe.tsx:50]: This doesn't handle the call to `setAuthoringSupported(!!info.features.authoredState)` that was deleted in this PR in this file:

lara-typescript/src/page-item-authoring/common/components/interactive-authoring.tsx

File: lara-typescript/src/page-item-authoring/common/components/interactive-authoring.tsx [hunk 2]
```diff
@@ -29,30 +29,11 @@ export const InteractiveAuthoring: React.FC<Props> = ({interactive, onAuthoredSt
...
   };
 
-  const handleHeightChange = (newHeight: number | string) => {
-    if (iframe.current) {
-      iframe.current.style.height = `${newHeight}px`;
-    }
-  };
-
-  const handleSupportedFeatures = (info: any) => {
-    setAuthoringSupported(!!info.features.authoredState);
```
Comment (dougmartin) [inline lara-typescript/src/page-item-authoring/common/components/interactive-authoring.tsx:39]: This call to `setAuthoringSupported` isn't handled in the new refactored code.

File: lara-typescript/src/page-item-authoring/common/components/interactive-iframe.tsx (hunk 1 above)
Comment (scytacki) [reply lara-typescript/src/page-item-authoring/common/components/interactive-iframe.tsx:51]: Good catch. next commit takes care of this.  However it looks like it was not being used, but we might use it in the future.

**Pull request merged**